timelock-impl/src/test/java/com/palantir/atlasdb/timelock/AsyncTimelockServiceImplTest.java

```diff
@@ -19,21 +19,180 @@
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 
+import com.codahale.metrics.MetricRegistry;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.palantir.atlasdb.futures.AtlasFutures;
+import com.palantir.atlasdb.timelock.api.GenericNamedMinTimestamp;
+import com.palantir.atlasdb.timelock.api.NamedMinTimestampLeaseResponse;
 import com.palantir.atlasdb.timelock.lock.AsyncLockService;
 import com.palantir.atlasdb.timelock.lock.LockLog;
+import com.palantir.atlasdb.timelock.lockwatches.BufferMetrics;
 import com.palantir.atlasdb.timelock.lockwatches.RequestMetrics;
+import com.palantir.timestamp.InMemoryTimestampService;
 import com.palantir.timestamp.ManagedTimestampService;
+import com.palantir.tritium.metrics.registry.DefaultTaggedMetricRegistry;
+import java.util.UUID;
+import java.util.concurrent.ScheduledExecutorService;
+import org.jmock.lib.concurrent.DeterministicScheduler;
+import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+public final class AsyncTimelockServiceImplTest {
+    private static final GenericNamedMinTimestamp TIMESTAMP_NAME = GenericNamedMinTimestamp.of("ToyTimestamp");
+
+    private final DeterministicScheduler executor = new DeterministicScheduler();
+    private final ManagedTimestampService timestampService = new InMemoryTimestampService();
+    private final AsyncLockService lockService = createLockService(executor);
+    private final AsyncTimelockService service = createAsyncTimelockService(timestampService, lockService);
 
-public class AsyncTimelockServiceImplTest {
     @Test
     public void delegatesInitializationCheck() {
         ManagedTimestampService mockMts = mock(ManagedTimestampService.class);
-        AsyncTimelockServiceImpl service = new AsyncTimelockServiceImpl(
+        AsyncTimelockServiceImpl timelockService = new AsyncTimelockServiceImpl(
                 mock(AsyncLockService.class), mockMts, mock(LockLog.class), mock(RequestMetrics.class));
         when(mockMts.isInitialized()).thenReturn(false).thenReturn(true);
 
-        assertThat(service.isInitialized()).isFalse();
-        assertThat(service.isInitialized()).isTrue();
+        assertThat(timelockService.isInitialized()).isFalse();
+        assertThat(timelockService.isInitialized()).isTrue();
+    }
+
+    // TODO(aalouane): enable when AsyncLockService implementation is merged
+    @Disabled
```

Comment:
I had ran this with everything wired in. I can wait for the PR with `AsyncLockService` functionality before merging here